lib/binary/ProofTraceParser.cpp

```diff
@@ -0,0 +1,97 @@
+#include <kllvm/binary/ProofTraceParser.h>
+
+namespace kllvm {
+
+#define INDENT_SIZE 2u
+
+void LLVMRewriteEvent::printSubstitution(
+    std::ostream &Out, unsigned indent) const {
+  std::string Indent(indent * INDENT_SIZE, ' ');
+  for (const auto &p : substitution) {
+    Out << Indent << p.first << " = kore[" << p.second.second << "]"
+        << std::endl;
```

Comment:
My preference for this kind of code - at least for the moment - is to continue _printing_ to an ostream, but to use `fmt` to do the actual formatting. Something like:
```c++
Out << fmt::format("{}{} = kore[{}]\n", Indent, p.first, p.second.second);
```